hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/checksum/ReconcileContainerTask.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.container.checksum;
+
+import org.apache.hadoop.ozone.container.ozoneimpl.ContainerController;
+import org.apache.hadoop.ozone.container.replication.AbstractReplicationTask;
+import org.apache.hadoop.ozone.protocol.commands.ReconcileContainerCommand;
+import org.apache.hadoop.util.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+/**
+ * Used to execute a container reconciliation task that has been queued from the ReplicationSupervisor.
+ */
+public class ReconcileContainerTask extends AbstractReplicationTask {
+  private final ReconcileContainerCommand command;
+  private final DNContainerOperationClient dnClient;
+  private final ContainerController controller;
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(ReconcileContainerTask.class);
+
+  public ReconcileContainerTask(ContainerController controller,
+      DNContainerOperationClient dnClient, ReconcileContainerCommand command) {
+    super(command.getContainerID(), command.getDeadline(), command.getTerm());
+    this.command = command;
+    this.controller = controller;
+    this.dnClient = dnClient;
+  }
+
+  @Override
+  public void runTask() {
+    long start = Time.monotonicNow();
+
+    LOG.info("{}", this);
+
+    try {
+      controller.reconcileContainer(dnClient, command.getContainerID(), command.getPeerDatanodes());
+      setStatus(Status.DONE);
+      long elapsed = Time.monotonicNow() - start;
+      LOG.info("{} completed in {} ms", this, elapsed);
```

Comment:
Yeah we need metrics specific to reconciliation tasks since the `ReplicationSupervisorMetrics` are for all replication tasks and we aren't using the `CommandHandler` metrics anymore for anything except invocation count. I filed HDDS-11345 for this.